premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs

```diff
@@ -147,4 +152,26 @@ public async Task DeletePreEnrollment(int preEnrollmentId, string studentEmail)
             throw new CoreException("Student cannot modify pre enrollment");
         return await _preEnrollmentRepository.GetRecommendationsForPreEnrollment(preEnrollmentId);
     }
+
+    public async Task<PreEnrollmentCourseParserOutput> StudentCompliesWithRequisites(int preEnrollmentId,
+        string studentEmail)
+    {
+        var preEnrollment = await ValidatePreEnrollmentExists(preEnrollmentId);
+        var selections = preEnrollment.Selections;
+        var student = await _studentValidationService.ValidateStudentExists(studentEmail, true);
+        var result = new PreEnrollmentCourseParserOutput();
+        foreach (var semesterOffer in selections)
+        {
+            var missingCourses = await _courseParsingService
+                .GetMissingCourses(student.CoursesTaken, semesterOffer.Course);
+            
+            if (missingCourses.Any())
+            {
+                result.NotComplyingCourses.Add(semesterOffer.Course);
```

Comment:
Maybe move this to an instance method for PreEnrollmentParserOutput. Would look something like:
```
var missingCourses = .... 
result.IncludeMissingCoursesFor(semesterOffer.course, missingCourses);
```